libraries/lib-ipc/IPCChannel.h

```diff
@@ -0,0 +1,49 @@
+#pragma once
+
+#include <cstddef>
+
+/**
+ * \brief Interface for sending data from client to server or vise versa,
+ * complemented by IPCChannelStatusCallback
+ */
+class IPC_API IPCChannel
+{
+public:
+   virtual ~IPCChannel();
+   /**
+    * \brief Write data to the channel
+    * \param bytes Pointer to the data
+    * \param length Number of bytes to send
+    */
+   virtual void Send(const void* bytes, size_t length) = 0;
+};
```

Comment:
When you need an abstract base class with just one virtual function besides the destructor, an alternative is just to generate the type you need from std::function:

```
using IPCChanel = std::function< void(const void *bytes, size_t length) >;
```

It's your choice whether you prefer that suggestion here.